### ISSUE TITLE
setting do_cic_particles = 1 and doPoissonSolve = 0 causes obscure error message

### ISSUE THREAD

**Issue by chongchonghe** [quokka-astro/quokka#832]
**Describe the bug**
When we set `do_cic_particles = 1` in a test problem (e.g., binary_orbit.cpp) and accidently set `doPoissonSolve_ = 0` (the default), we get this error which is hard to debug:
```
0::Assertion `i < (this->std::vector<T, Allocator>::size())' failed, file "/Users/cche/softwares/quokka/quokka.copy3/extern/amrex/Src/Base/AMReX_Vector.H", line 35 !!!
```

**To Reproduce**
1. Compile this problem 'BinaryOrbitCIC' whiling setting `sim.doPoissonSolve_ = 0;`.
2. Run this problem
3. See error

**Suggestion**
This error occurs because the code tries to access `phi` in `kickParticlesAllLevels()` but `phi` is not initialized unless the Poisson solver is turned on. So, my suggestion is to abort the program if `do_cic_particles = 1 && doPoissonSolve_ = 0`. With the new particle interface, we should abort the program if we turn on any of the particle types with gravity and do not turn on Poisson solver. 

**Additional context**

**Comment by markkrumholz** [comment]
Agree that that is a useful check to implement in the new particle architecture. Same thing for particle types that emit radiation and in simulations that are run with radiation turned off.

**Comment by BenWibking** [comment]
Yes, I agree. Feel free to add a PR with this check and an informative error message.

**Comment by BenWibking** [comment]
Since it doesn't actually cause wrong results or a failure in a case where it should work, I'm going to take off the red 'bug' label. But it should be more informative to the user, so I've instead added the yellow 'bug: usability' label.
